src/haven/motor_position.py

```diff
@@ -73,24 +89,18 @@ def default_collection():
 
 
 # Prepare the motor positions
-def rbv(motor):
+async def rbv(motor):
     """Helper function to get readback value (rbv)."""
-    try:
-        # Wrap this in a try block because not every signal has this argument
-        motor_data = motor.get(use_monitor=False)
-    except TypeError:
-        log.debug("Failed to do get() with ``use_monitor=False``")
-        motor_data = motor.get()
-    if hasattr(motor_data, "readback"):
-        return motor_data.readback
-    elif hasattr(motor_data, "user_readback"):
-        return motor_data.user_readback
+    if hasattr(motor, "readback"):
+        return await motor.readback.get_value()
+    elif hasattr(motor, "user_readback"):
+        return await motor.user_readback.get_value()
     else:
-        return motor_data
+        return await motor.get_value()
 
 
-def save_motor_position(*motors, name: str, collection=None):
-    """Save the current positions of a number of motors to a database.
+def save_motor_position(*motors, name: str, md: Mapping = {}):
```

Comment:
I used to do something like:

```python
def save_motor_position(*motors, name: str, md: Mapping = None):
    if md is None:
        md == {}
```
which has a few drawbacks:

1. It's unclear from the call signature what the default argument will be.
2. More lines of code that don't really do anything.
3. More complicated type hints since we need ``Mapping | None`` to account for a value we'll never actually use.

If the default argument really is only knowable at runtime, then there's not a better option. But that's not true for ``md``. I switched to just including ``{}`` as the default because I decided that mutating arguments is the real anti-pattern, not using a mutable default argument. Instead, I just don't ever mutate arguments unless I explicitly describe it in the docstring.

What do you think about that explanation? I can always go back to ``md: Mapping | None = None`` if there's a benefit you can think of, or if it turns out I'm mutating ``md`` somewhere.

Reply:
Or we could use a frozendict as the default.

Reply:
What you suggested makes sense, so I will resolve it